File: app/src/main/java/com/nelyan/adapter/DetailsUpcomingAdapter.java
package com.nelyan.adapter;

import android.app.Activity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.FragmentActivity;
import androidx.recyclerview.widget.RecyclerView;

import com.nelyan.R;
public class DetailsUpcomingAdapter extends RecyclerView.Adapter {
    Activity a;

    public DetailsUpcomingAdapter(FragmentActivity activityDetailsActivity) {
        this.a=activityDetailsActivity;
    }

    public class RecyclerViewHolder extends RecyclerView.ViewHolder {
        public RecyclerViewHolder(View view) {
            super(view);

        }
    }
    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(a).inflate(R.layout.row_upcoming_details,parent,false);
        DetailsUpcomingAdapter.RecyclerViewHolder viewHolder = new DetailsUpcomingAdapter.RecyclerViewHolder(v);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {

    }

    @Override
    public int getItemCount() {
        return 2;
    }
}

File: app/src/main/java/com/nelyan/adapter/DetailsUpcomingAdapter.kt
package com.nelyan.adapter

import android.app.Activity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.FragmentActivity
import androidx.recyclerview.widget.RecyclerView
import com.nelyan.R

class DetailsUpcomingAdapter(activityDetailsActivity: FragmentActivity) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    var a: Activity

    inner class RecyclerViewHolder(view: View?) : RecyclerView.ViewHolder(view!!)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val v = LayoutInflater.from(a).inflate(R.layout.row_upcoming_details, parent, false)
        return RecyclerViewHolder(v)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {}
    override fun getItemCount(): Int {
        return 2
    }

    init {
        a = activityDetailsActivity
    }
}